src/example/incompressible_navier_stokes/incompressible_navier_stokes.jl

```diff
@@ -0,0 +1,276 @@
+
+    M1 = assemble_bilinear(m1, U_vel, V_vel)
+    M0 = copy(M1)
+    Bcube.apply_dirichlet_to_matrix!(M1, U_vel, V_vel, mesh)
+
+    A2 = assemble_bilinear(a2, U_pre, V_pre)
+    Bcube.apply_dirichlet_to_matrix!(A2, U_pre, V_pre, mesh)
+
+    time = 0.0
+    itime = 0
+
+    while time <= finalTime
+        time += Δt
+        itime += 1
+
+        println("Time stepping : time = ", time, " / ", finalTime)
+
+        # assemble l1
+        L1 = assemble_linear(l1, V_vel)
+        Wd = Bcube.assemble_dirichlet_vector(U_vel, V_vel, mesh, time)
+        # Apply lift
+        L1 = L1 - M0 * Wd
+
+        # Apply homogeneous dirichlet condition
+        Bcube.apply_homogeneous_dirichlet_to_vector!(L1, U_vel, V_vel, mesh)
+
+        # Compute tentative velocity
+        sol = M1 \ L1
+
+        set_dof_values!(velocity, sol .+ Wd)
+
+        # Assemble l2
+        L2 = assemble_linear(l2, V_pre)
+        Bcube.apply_homogeneous_dirichlet_to_vector!(L2, U_pre, V_pre, mesh)
+
+        # Compute pressure
+        sol = A2 \ L2
+
+        set_dof_values!(pressure, sol)
+
+        # Assemble l3
+        L3 = assemble_linear(l3, V_vel)
+        # Apply lift
+        L3 = L3 - M0 * Wd
+
+        # Apply homogeneous dirichlet condition
+        Bcube.apply_homogeneous_dirichlet_to_vector!(L3, U_vel, V_vel, mesh)
+
+        # Velocity correction step
+        sol = M1 \ L3
+
+        set_dof_values!(velocity, sol .+ Wd)
+
+        # Write outputs
+        if itime % 100 == 0
+            vars = Dict("Velocity" => velocity, "Pressure" => pressure)
+            Bcube.write_vtk_lagrange(
+                joinpath(outputpath, "output_projection"),
+                vars,
+                mesh,
+                itime,
+                time,
+            )
+        end
+    end
+end
+# The animation below shows the result of simulation with the projection method:
+# ![](../assets/navier_stokes_projection.gif)
+
+# Function that solves the problem using a mixed formalism
+function run_unsteady_mixed()
+    # read mesh
+    mesh = read_msh(meshpath)
+
+    # Definition of trial and test function spaces
+    fsu = FunctionSpace(fspace, degree_u)
+    U_vel = TrialFESpace(
+        fsu,
+        mesh,
+        Dict(
+            "left" => inlet_velocity,
+            "top" => SA[0.0, 0.0],
+            "bottom" => SA[0.0, 0.0],
+            "cylinder" => SA[0.0, 0.0],
+        );
+        size = 2,
+    )
+    V_vel = TestFESpace(U_vel)
+
+    fsp = FunctionSpace(fspace, degree_p)
+    U_pre = TrialFESpace(fsp, mesh; size = 1)
+    V_pre = TestFESpace(U_pre)
+
+    U = MultiFESpace(U_vel, U_pre)
+    V = MultiFESpace(V_vel, V_pre)
+
+    velocity = FEFunction(U_vel, 0.0)
+    pressure = FEFunction(U_pre, 0.0)
+
+    # Define measures for cell
+    dΩ = Measure(CellDomain(mesh), degquad)
+
+    # definition of bilinear and linear forms
+    m((u, p), (v, q)) = ∫(u ⋅ v)dΩ
+    a((u, p), (v, q)) = ∫(ν * ∇(u) ⊡ ∇(v) - tr(∇(v)) * p + tr(∇(u)) * q)dΩ
```

Comment:
Please use composition for complex weak forms

Reply:
I tried writting it like this:
```Julia
function a((u, p), (v, q))
    f(∇u, ∇v, p, q) = ν * ∇u ⊡ ∇v - tr(∇v) * p + tr(∇u) * q
    ∫(f ∘ (∇(u), ∇(v), p, q))dΩ
end
```
but I must be doing something wrong because I get an error message (no method matching length(::Bcube.LazyOperators.LazyOperator{typeof(*), Tuple{…}}))

Reply:
maybe comment two of the three sum operands to identify the one causing the problem